core/project.cpp

```diff
@@ -62,19 +62,22 @@ ProjectStatus::Status ProjectStatus::projectStatus( const std::shared_ptr<Projec
     return ProjectStatus::NoVersion;
   }
 
-  //
-  // TODO: this check for local modifications should be revisited
-  //
-
   // Something has locally changed after last sync with server
   QString metadataFilePath = project->local->projectDir + "/" + MerginApi::sMetadataFile;
   QDateTime lastModified = CoreUtils::getLastModifiedFileDateTime( project->local->projectDir );
-  QDateTime lastSync = QFileInfo( metadataFilePath ).lastModified();
+  QDateTime lastSync = QFileInfo( metadataFilePath ).lastModified().toUTC();
   MerginProjectMetadata meta = MerginProjectMetadata::fromCachedJson( metadataFilePath );
   int filesCount = CoreUtils::getProjectFilesCount( project->local->projectDir );
   if ( lastSync < lastModified || meta.files.count() != filesCount )
   {
-    return ProjectStatus::Modified;
+    // When GPKG is opened, its header is updated and therefore lastModified timestamp is updated as well.
+    // Double check if there is really something to upload
+    QList<MerginFile> localFiles = MerginApi::getLocalProjectFiles( project->local->projectDir + "/" );
+    MerginProjectMetadata serverProject = MerginProjectMetadata::fromCachedJson( metadataFilePath );
+    ProjectDiff diff = MerginApi::compareProjectFiles( serverProject.files, serverProject.files, localFiles, project->local->projectDir );
```

Comment:
What if gives you error?

Reply:
Error is handled by GeoDiffUtils.
`MerginApi::compareProjectFiles` function is using `GeodiffUtils::hasPendingChanges` function, which in case of error returns true - in-spite of fact that something went wrong, it assumes the file has changed (this is from code/comment). So in the case of error, diff may contain some local changes and therefore the returned status is `Modified`.
The same function and logic is used in MerginApi sync.